### PR TITLE
Libvirt live migration test implementation.

### PR DESCRIPTION
**Pull request by vlad-nikityuk**
- Restructured code, moved openstack specific code to `openstack` folder, added `commons` and `libvirt` folders.
- Added config model and parser for libvirt test. Changed config structure as compared with openstack test config.
- Added ansible scripts for libvirt specific steps.

⚠️ _Current libvirt and openstack test implementations have less shared code as it might be.
Since things may change and code base is not stable, my proposal to leave it as is for now._

### CODE REVIEW

File: src/main/scala/org/orchestra/common/CallbackActor.scala [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+
+object CallbackActor {
+  def props(callback: Callback, reaper: ActorRef): Props =
+    Props(new CallbackActor(callback.host, callback.port, reaper))
+}
+
+class CallbackActor(host:String, port:Int, reaper: ActorRef) extends Actor {
+
+//  var callback_listener: ActorRef = null
```
Comment (vlad-nikityuk) [inline src/main/scala/org/orchestra/common/CallbackActor.scala:20]: uncomment or remove

File: src/main/scala/org/orchestra/common/CallbackActor.scala [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  val callback_service = context.actorOf(InstanceCallbackService.props(context.self), name="cb_service")
+  reaper ! WatchClient(callback_service)
+
+  implicit val system = context.system
+  IO(Http) ! Http.Bind(callback_service, interface="0.0.0.0", port=port)
+
+
+
+  override def receive: Receive = {
+//    case "wait" => Thread.sleep(1000); self ! InstanceAvailableEvent("")
```
Comment (vlad-nikityuk) [inline src/main/scala/org/orchestra/common/CallbackActor.scala:30]: uncomment or remove

File: src/main/scala/org/orchestra/common/CallbackActor.scala [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    case x: InstanceAvailableEvent => context.parent ! "processNextEvent"
+
+    case Http.Bound(address) => {
+      context.system.log.debug("http bound event received: {}", address)
+      val callback_listener = sender()
+      reaper ! WatchClient(callback_listener)
+    }
+//    case Http.Unbound => {
```
Comment (vlad-nikityuk) [inline src/main/scala/org/orchestra/common/CallbackActor.scala:38]: uncomment or remove

File: src/main/scala/org/orchestra/common/CallbackActor.scala [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+//      callback_listener ! PoisonPill
+//    }
+  }
+
+  override def postStop = {
+//    if(callback_listener != null)
```
Comment (vlad-nikityuk) [inline src/main/scala/org/orchestra/common/CallbackActor.scala:44]: uncomment or remove

**Review by timofei-durakov — APPROVED** [review]
lgtm

**Pull request merged**